### PR TITLE
AO-26:Create A List of RefApp 2.10 modules only for openmrs AddonsIndex 

### PR DESCRIPTION
**Pull request by sherrif10**
**What i did**
 Have only added module modules for RefApp 2.10 to be included in openmrs Addon index.
Didnot include chart search module among modules because it was removed from refapp 2.10 release

Issue WorkedOn
https://issues.openmrs.org/browse/AO-26

cc @dkayiwa

### CODE REVIEW

**Comment by HerbertYiga** [comment]
well done on this sharif,@sharif you also forgot to add a ticket Id on your commit

**Comment by sherrif10** [comment]
Ooops thnks herbert for that catch

**Comment by sherrif10** [comment]
cc @dkayiwa  any response on this

**Comment by dkayiwa** [comment]
@sherrif10 did you respond to Herbert’s review?

**Comment by sherrif10** [comment]
Finished thanks

**Comment by dkayiwa** [comment]
https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by sherrif10** [comment]
Haa seems i have reserved all pull request tips, May be you can point to me what am missing interesting

**Comment by dkayiwa** [comment]
@sherrif10 cross check.

**Comment by sherrif10** [comment]
@dkayiwa  still missing something??

**Comment by dkayiwa** [comment]
Oh yes.

**Comment by sherrif10** [comment]
which is that please

**Comment by dkayiwa** [comment]
Very simple, just go through it again and for each tip ask yourself, did i do this?

**Comment by sherrif10** [comment]
There seem to be not any missing point among pull request tips that i have not addressed

**Comment by dkayiwa** [comment]
@sherrif10 just type for me here a comma separated list of the titles in Bold, on the tips page, that you did.

**Comment by sherrif10** [comment]
My title means the list of modules indicated in json  i have only one title

**Comment by dkayiwa** [comment]
@sherrif10 i mean the titles in bold on this page: https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by sherrif10** [comment]
**Got ticket?,Branch name == ticket name,Reference the ticket,Pull against master.,Update first,Tests passing?,Follow our coding conventions.,Avoid a noisy pull request. ,Squash!,Help us help you,JIRA workflow,Use JIRA.,Avoid multiple pull requests for the same ticket,Replacing a pull request.,Abandoning a ticket,Travis CI build failures.**

**Comment by dkayiwa** [comment]
Can you now edit the above and put what you did in brackets? For instance: **Got ticket**(Ticket AO-26), **Branch name**(The name of the branch), **Reference the ticket** (show where that is), etc

**Comment by sherrif10** [comment]
@dkayiwa: All that have been resolved,Are you really sure i havent  i have included the ticket id above among the description on my very first comment on this page, i created a branch above branch Name- AO-26, i also shared the pull request ticket link   i dont know what you are trying to mean regards to this ticket

**Comment by sherrif10** [comment]
Am not understanding your point, Everything you pointed above has been added except if am mistaken may be.

**Comment by dkayiwa** [comment]
For each of the titles you put above in bold, can you put text in brackets for what you did to meet what is talked about on the tips page? Just like the example that i gave.

**Comment by sherrif10** [comment]
i did this because you see where i reported all the comment though that's not how we report tickets, so am changing the soon after ur comment

File: src/main/resources/add-ons-to-index.json [hunk 1]
```diff
@@ -6750,6 +6750,177 @@
...
+          "uid": "org.openmrs.module.calculation",
+          "version": "1.2"
+        },
+        {
+          "uid": "org.openmrs.module.core-apps-module",
+          "version": "1.28.0"
+        },
+        {
+          "uid": "org.openmrs.module.data-exchange-module",
+          "version": "1.3.3"
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6800]: Is the above version correct?

File: src/main/resources/add-ons-to-index.json [hunk 2]
```diff
@@ -6750,6 +6750,177 @@
...
+        },
+        {
+          "uid": "org.openmrs.module.emrapi",
+          "version": "1.28.0"
+        },
+        {
+          "uid": "org.openmrs.module.event",
+          "version": "2.8.0"
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6808]: Is the above version correct?

File: src/main/resources/add-ons-to-index.json [hunk 3]
```diff
@@ -6750,6 +6750,177 @@
...
+        },
+        {
+          "uid": "org.openmrs.module.openmrs-fhir-module",
+          "version": "1.19.0"
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6812]: Is the above version correct?

File: src/main/resources/add-ons-to-index.json [hunk 4]
```diff
@@ -6750,6 +6750,177 @@
...
+        {
+          "uid": "org.openmrs.module.metadatamapping",
+          "version": "1.3.4"
+        },
+        {
+          "uid": "org.openmrs.module.metadatasharing",
+          "version": "1.6.0"
+        },
+        {
+          "uid": "org.openmrs.module.namephonetics",
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6851]: Are you sure this module is included?

File: src/main/resources/add-ons-to-index.json [hunk 5]
```diff
@@ -6750,6 +6750,177 @@
...
+          "uid": "org.openmrs.module.registration-app-module",
+          "version": "1.16.0"
+        },
+        {
+          "uid": "org.openmrs.module.registration-core-module",
+          "version": "1.9.0"
+        },
+        {
+          "uid": "org.openmrs.module.reporting",
+          "version": "1.19.0"
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6884]: Is the above version correct?

File: src/main/resources/add-ons-to-index.json [hunk 6]
```diff
@@ -6750,6 +6750,177 @@
...
+          "uid": "org.openmrs.module.serialization-xstream",
+          "version": "0.2.14"
+        },
+        {
+          "uid": "org.openmrs.owa.sysadmin",
+          "version": "1.2"
+        },
+        {
+          "uid": "org.openmrs.module.uicommons",
+          "version": "2.11.0"
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6908]: Is the above version correct?

File: src/main/resources/add-ons-to-index.json [hunk 7]
```diff
@@ -6750,6 +6750,177 @@
...
+        },
+        {
+          "uid": "org.openmrs.module.uiframework",
+          "version": "3.16.0"
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6912]: Is the above version correct?

File: src/main/resources/add-ons-to-index.json [hunk 8]
```diff
@@ -6750,6 +6750,177 @@
...
+        },
+        {
+          "uid": "org.openmrs.module.uilibrary",
+          "version": "2.0.6"
+        },
+        {
+          "uid": "org.openmrs.module.webservices-rest",
+          "version": "2.26.0"
```
Comment (dkayiwa) [inline src/main/resources/add-ons-to-index.json:6920]: Is the above version correct?

**Comment by dkayiwa** [comment]
Are you sure that you have included all the necessary modules?

File: src/main/resources/add-ons-to-index.json (hunk 1 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved thanks

File: src/main/resources/add-ons-to-index.json (hunk 2 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved thanks

File: src/main/resources/add-ons-to-index.json (hunk 3 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved thanks

File: src/main/resources/add-ons-to-index.json (hunk 4 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved, this was removed but maintained its version in distro

File: src/main/resources/add-ons-to-index.json (hunk 5 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved thanks

File: src/main/resources/add-ons-to-index.json (hunk 6 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved thanks

File: src/main/resources/add-ons-to-index.json (hunk 7 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved thanks

File: src/main/resources/add-ons-to-index.json (hunk 8 above)
Comment (sherrif10) [reply src/main/resources/add-ons-to-index.json]: Resolved thanks alot for your time

**Comment by sherrif10** [comment]
i have now  finished updated the file, The challenge was that i was following this[ draft here](https://wiki.openmrs.org/display/RES/Reference+Application+2.10+Release+Issue+Tracking) and compare versions with those on addons so that's where i messed up with version . thanks for your review

**Comment by sherrif10** [comment]
cc @dkayiwa  kindly take another look thanks

**Comment by dkayiwa** [comment]
@sherrif10 how many modules do you have in here?

**Comment by sherrif10** [comment]
40 modules

**Comment by dkayiwa** [comment]
@sherrif10 i mean modules without owas.

**Comment by sherrif10** [comment]
They are  only 39

**Comment by dkayiwa** [comment]
@sherrif10 how many are they expected to be?

**Comment by sherrif10** [comment]
They are expected to be 40 modules

**Comment by sherrif10** [comment]
cc @dkayiwa  any more suggestions on this

**Comment by dkayiwa** [comment]
@sherrif10 is 39 = 40?

**Comment by sherrif10** [comment]
@dkayiwa: True we only have 39 modules because we removed namephonetics

**Comment by dkayiwa** [comment]
Is 39 = 40 and you say `True`?

**Comment by sherrif10** [comment]
No we needed to have only 39 modules not 40

**Comment by dkayiwa** [comment]
How many modules do we have here? https://qa-refapp.openmrs.org/openmrs/admin/modules/module.list

**Comment by sherrif10** [comment]
so your collect answer is No thnks

**Comment by sherrif10** [comment]
There are 39 modules only

**Comment by dkayiwa** [comment]
Are you sure?

**Comment by sherrif10** [comment]
Sorry they are 40 modules here
https://qa-refapp.openmrs.org/openmrs/admin/modules/module.list

**Comment by dkayiwa** [comment]
And so why are you having 39?

**Comment by sherrif10** [comment]
i have 40 modules sorry in my pr ,so meaning if we had not removed
namephonetics they have been 41 sorry for that

**Comment by dkayiwa** [comment]
As i said before above, modules without owas? Don't you think you should get some rest and come back tomorrow when you have a fresh mind?

**Comment by sherrif10** [comment]
Sorry that accounts to 39 modules in the pr without owas

**Comment by dkayiwa** [comment]
I again ask, How many modules do we have here? https://qa-refapp.openmrs.org/openmrs/admin/modules/module.list

**Comment by sherrif10** [comment]
39 modules without owas

**Comment by dkayiwa** [comment]
Does that page list owas or modules?

**Comment by sherrif10** [comment]
Yes owas is among the 40 modules in the module list

**Comment by sherrif10** [comment]
Let me hope by owas we mean open web apps module

**Comment by dkayiwa** [comment]
Is what you have in this pull request the same as what you have on that page?

**Comment by sherrif10** [comment]
No because this pr is for those modules for refApp 2.10

**Comment by sherrif10** [comment]
however same modules that are in the page but different versions

**Comment by dkayiwa** [comment]
What is on that page that does not belong to refapp 2.10?

**Comment by dkayiwa** [comment]
Am not talking of versions. Am talking of the actual modules.

**Comment by sherrif10** [comment]
True modules are the same as the ones i have in the pull request

**Comment by dkayiwa** [comment]
Are you sure?

**Comment by sherrif10** [comment]
i only don't have fhir2 module because it was not yet updated, we used fhir
module instead of fhir2

**Comment by dkayiwa** [comment]
Is that the only difference?

**Comment by sherrif10** [comment]
Another difference is  openmrs.owa.sysadmin that i have in pr and not in
the module list

**Comment by dkayiwa** [comment]
Are all modules on that page also in your pull request?

**Comment by sherrif10** [comment]
No thanks

**Comment by dkayiwa** [comment]
Which ones are not?

**Comment by sherrif10** [comment]
openmrs-owa-sysadmin and fhir2 , i have openmrs-owa-sysadmin in the pull
request but not in the module list and i dont have fhir2 module in the pr
but it's among in the module list

**Comment by dkayiwa** [comment]
I mean, what is on that module list page but is not in your pull request.

**Comment by sherrif10** [comment]
Fhir2 module module

**Comment by dkayiwa** [comment]
fhir and fhir2 are the same. So what is on that module list page but is not in your pull request

**Comment by sherrif10** [comment]
Nothing everything seem equal

**Comment by dkayiwa** [comment]
How many modules do you have on the modules list page?

**Comment by sherrif10** [comment]
38 modules

**Comment by dkayiwa** [comment]
Are you sure that those are the number of modules here? https://qa-refapp.openmrs.org/openmrs/admin/modules/module.list

**Comment by sherrif10** [comment]
sorry 39 modules   cant see which module is missing

**Comment by dkayiwa** [comment]
Are you sure that https://qa-refapp.openmrs.org/openmrs/admin/modules/module.list contains 39 modules only?

**Comment by sherrif10** [comment]
without owa module they are 39 modules

**Comment by dkayiwa** [comment]
Everything there is a module. Owas are here: https://qa-refapp.openmrs.org/openmrs/module/owa/manage.form

**Comment by sherrif10** [comment]
they are 39 modules without owas

**Comment by dkayiwa** [comment]
I said that there is not a single owa on the page: https://qa-refapp.openmrs.org/openmrs/admin/modules/module.list 
Everything you see there is a module. 

If you are looking for owas, they are here: https://qa-refapp.openmrs.org/openmrs/module/owa/manage.form

Does it make sense?

**Comment by sherrif10** [comment]
yes sure makes sense thanks

**Comment by dkayiwa** [comment]
So i ask for the last time, how many modules are on this page? https://qa-refapp.openmrs.org/openmrs/admin/modules/module.list

**Comment by sherrif10** [comment]
40modules

**Comment by dkayiwa** [comment]
And how many are in your pull request?

**Comment by sherrif10** [comment]
39modules

**Comment by dkayiwa** [comment]
So which module is on the modules page, but not in your pull request?

**Comment by sherrif10** [comment]
appui module thnks

**Comment by dkayiwa** [comment]
Now take a second look at Reuben's original pull request to see if yours has all that it should.

**Comment by sherrif10** [comment]
Sure it has all that it should be having according to this[ Pr](https://github.com/openmrs/openmrs-contrib-addonindex/pull/71/files) everything look similar the only difference is package.lock.json file where by its already up to date

**Comment by dkayiwa** [comment]
Are you sure?

**Comment by sherrif10** [comment]
Sure everything seems to be same except package.lock.json where i think no additionals

**Comment by dkayiwa** [comment]
That is not all. Please take another careful look.

**Comment by sherrif10** [comment]
@dkayiwa i have checked all the files in package.lock.json and it seem all was fixed may be if its out of that file

**Comment by dkayiwa** [comment]
I mean add-ons-to-index.json

**Comment by sherrif10** [comment]
it have namephonetics module , i think we should have namephonetics , its very confusing though

**Comment by dkayiwa** [comment]
Is that all that Reuben's file has that is not in yours?

**Comment by sherrif10** [comment]
yes according to this file here https://github.com/openmrs/openmrs-contrib-addonindex/pull/71/files#

**Comment by dkayiwa** [comment]
Are you sure?

**Comment by sherrif10** [comment]
There is also chart search module and namephonetic s those were the ones missing in my pr

**Comment by dkayiwa** [comment]
Is that all?

**Comment by sherrif10** [comment]
i think yes

**Comment by dkayiwa** [comment]
Are you sure?

**Comment by sherrif10** [comment]
yes have checked again

**Comment by dkayiwa** [comment]
You just need to do it systematically. For each item that you see in the original pull request, carefully check if it exists in your pr.

**Comment by sherrif10** [comment]
It wa  openmrs owa sysadmin i had removed it last night sorry i reverted it back

**Comment by dkayiwa** [comment]
Any reason why you have added namephonetics and chartsearch?

**Comment by sherrif10** [comment]
Because they were part of  last refApp 2.9 addons according to raubens [commit](https://github.com/openmrs/openmrs-contrib-addonindex/pull/71/files#)

**Comment by dkayiwa** [comment]
Were they part of the refapp 2.10 release?

**Comment by sherrif10** [comment]
Nope actually they were not part of refapp 2.10 release

**Comment by sherrif10** [comment]
Let me remove chartsearch module because this is neither bundled in refApp however am not sure whether we need to only have specifically those modules for refapp,

**Comment by sherrif10** [comment]
cc @dkayiwa  any more review on this, seems we have modules in reference to those that were in refApp 2.10

**Comment by dkayiwa** [comment]
Are you sure that the modules contained here are what you have? https://demo.openmrs.org/openmrs/admin/modules/module.list

**Comment by sherrif10** [comment]
cc @dkayiwa  yes now its updated

**Comment by sherrif10** [comment]
Thanks

**Pull request merged**